### DISCUSSION TITLE
How do I diffuse a backbone of n length around an active site?

### DISCUSSION THREAD

**Discussion by danielguion** [RosettaCommons/RFdiffusion#180] [Q&A]
Hello,

I am working on a project where I need to generate protein backbones to facilitate the geometry of a specific active site. The active site is represented by 7 residues, and I have a requirement to create backbones of different lengths (e.g., 40, 50, 60 amino acids).

Here's the part of my current code for generating the protein with the active site model, which is not doing what I want:

`./scripts/run_inference.py 'contigmap.contigs=[B11-11/1-10/B28-28/1-10/B39-39/1-10/B48-48/1-10/B52-52/1-10/B84-84/1-10/B88-88]' inference.input_pdb=/data/sequences/.pdb inference.ckpt_override_path=models/ActiveSite_ckpt.pt inference.num_designs=1 inference.output_prefix=outputs/`

My goal is to generate proteins where the length of amino acid sequences between the specified active site residues is random. I want to control the overall length of the protein but not the distribution of amino acids between the active site residues.

Can anyone suggest modifications to my code or a different approach to achieve this? Specifically, I am looking for a way to:

- Randomize the length of sequences between the active site residues.
- Control the overall length of the protein backbone.

Any help or guidance would be greatly appreciated.

Thank you!

**Comment by roccomoretti** [comment]
The syntax you use (the `/1-10/`) should cause RFdiffusion to randomly choose a length for each segment. These segment lengths should be different each time you run RFdiffusion.

If you want to make sure that the overall length stays in a particular range, you can use `contigmap.length` (e.g. `contigmap.length=80-100` or `contigmap.length=120-120`) to make sure the randomly chosen lengths sum up to the proper overall length.
